### ISSUE TITLE
Hunters Not Spawning or Staying

### ISSUE THREAD

**Issue by princessbinas** [TeamLapen/Vampirism#195]
## Versions

    Minecraft: 1.11.2
    Forge: 13.20.0.2265
    Vampirism: Vampirism-1.11.2-1.2.0-alpha+06-10-15-15 and Vampirism-1.11.2-1.2.0-beta.1

## Issue Description

In single player, I decided to use a bit of creative mode to create a base and improve a village for hunter farming. I have dropped vampires and villagers near the tents I placed or rearranged and not once have I seen a hunter spawned. I even dropped hunters on top of the tents and that did not work either.

When I grew frustrated with hunters not spawning on their own, I threw some spawn eggs for hunters and advanced hunters. About two to five minutes later, when I checked back, they had despawned. I tried sticking a villager and some vampires in front of them but that did not make them stay. They still despawned while the villager that came from the villager eggs I used stayed (and a bit crowded).

## Reproduce Steps

Just duplicate the versions I am using and follow what I do in the description.

**Comment by maxanier** [comment]
Manually placed tents do not spawn hunters.  And any hunters that are not in a village or spawned by a tent will despawn like most vanilla entities.

However, there is a test command you can use to make the tent spawn hunters if I remember correctly. Look at the tent and use '/vtest tent'

**Comment by princessbinas** [comment]
I have tried /vampirism-test tent (which is the command you are talking about), and it only makes the hunters spawn once and only once, even when there is a village right next to the tents or on top of them. The only thing that seems to get any results is using a command to get a mob spawner and right clicking it with the hunter spawn eggs.

**Comment by maxanier** [comment]
Sorry, always forget that `vtest` is `vampirism-test` outside of dev environment.

The tent should try to keep around 2-3 hunters in a small area (12x12 I think). If you kill all of them, new ones should spawn eventually, don't they?

I could add a second tent item to the creative tab that's automatically enables spawning. What do you think?

**Comment by princessbinas** [comment]
They don't respawn upon all of them dying and the area I made for them is a little bit bigger to keep them contained.

That sounds like a good idea.

**Comment by maxanier** [comment]
I've tested it myself and they are respawning in my world. Then your world has reached the hunter entity cap I guess.
To make there are not too many hunters in the world, the tent checks how many hunters are loaded before spawning (like vanilla does for monsters etc).
The current CAP is 25 or 30 if there is one player online.

If it is a test world you could run '/kill @e[type=vampirism:vampire_hunter]'.

If you want more hunter in your world, you have to use a regular spawner or some other method.

**Comment by maxanier** [comment]
https://minecraft.curseforge.com/projects/vampirism-become-a-vampire/files/2443231

**Comment by maxanier** [comment]
Can this be closed or is there still an issue?

**Comment by maxanier** [comment]
I actually might have found another issue related to this. #210
Did you reload the world after placing the tent?
Tents do not work anymore after reloading. (Stupid mistake I made)

I just never reloaded the world between testing so never noticed that
